File: BotBuilderNodeJs/libraries/botframework-schema/src/sharepoint/submitAction.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { IAction } from './IAction';
import { ConfirmationDialog } from './confirmationDialog';

/**
 * Sharepoint action.submit 
 */
export class SubmitAction implements IAction{
    private type: string = 'Submit';
    private parameters: { [key: string] : unknown };
    private confirmationDialog: ConfirmationDialog;

    /**
     * Initializes a new instance of the SubmitAction class
     */
    public SubmitAction() {
        // Do nothing
    }

    /**
     * Sets parameters property of type key:value pair
     */
    public set Parameters (parameters: { [key: string] : unknown }) {
        this.parameters = parameters;
    }

    /**
     * Gets parameters property of type key:value pair
     */
    public get Parameters(): { [key: string] : unknown } {
        return this.parameters; 
    }

    /**
     * Sets confirmation dialog property of type ConfirmationDialog
     */
    public set ConfirmationDialog (confirmationDialog: ConfirmationDialog) {
        this.confirmationDialog = confirmationDialog;
    }

    /**
     * Gets confirmation dialog property of type ConfirmationDialog
     */
    public get ConfirmationDialog(): ConfirmationDialog {
        return this.confirmationDialog; 
    }
}

File: BotBuilder DotNet/libraries/Microsoft.Bot.Schema/SharePoint/SubmitAction.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;

namespace Microsoft.Bot.Schema.SharePoint
{
    /// <summary>
    /// Action.Submit
    /// </summary>
    public class SubmitAction: Action
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="SubmitAction"/> class.
        /// </summary>
        public SubmitAction()
        {
            // Do nothing
        }

        /// <summary>
        /// Gets or Sets the action parameters of type <see cref="IDictionary<string, Object>"/>.
        /// </summary>
        /// <value>This value is the parameters of the action.</value>
        [JsonProperty(PropertyName = "parameters")]
        public IDictionary<string, Object> Parameters { get; set; }
    
        /// <summary>
        /// Gets or Sets confirmation dialog associated with this action of type <see cref="ConfirmationDialog"/>.
        /// </summary>
        /// <value>This value is the confirmation dialog associated with this action</value>
        [JsonProperty(PropertyName = "confirmationDialog")]
        public ConfirmationDialog ConfirmationDialog { get; set; }
    }
}
